packages/inter-protocol/src/proposals/replaceElectorate.js

```diff
@@ -289,33 +289,87 @@ const startNewEconomicCommittee = async (
  * Starts a new Economic Committee Charter by creating an instance with the
  * provided committee specifications.
  *
- * @param {EconomyBootstrapPowers} powers - The resources and capabilities
- *   required to start the committee.
+ * @param {EconomyBootstrapPowers &
+ *   PromiseSpaceOf<{ retiredContractInstances: MapStore<string, Instance> }>} powers
+ *   - The resources and capabilities required to start the committee.
+ *
  * @returns {Promise<EconCharterStartResult>} A promise that resolves to the
  *   charter kit result.
  */
 const startNewEconCharter = async ({
-  consume: { startUpgradable },
-  produce: { econCharterKit },
+  consume: {
+    board,
+    startUpgradable,
+    contractKits: contractKitsP,
+    econCharterKit: econCharterKitP,
+    retiredContractInstances: retiredContractInstancesP,
+  },
+  produce: {
+    econCharterKit: produceEconCharterKit,
+    retiredContractInstances: produceRetiredInstances,
+  },
   installation: {
     consume: { binaryVoteCounter: counterP, econCommitteeCharter: installP },
   },
   instance: {
     produce: { econCommitteeCharter },
+    consume: { econCommitteeCharter: previousInstanceP },
   },
 }) => {
-  const [charterInstall, counterInstall] = await Promise.all([
+  const [
+    charterInstall,
+    counterInstall,
+    previousInstance,
+    contractKits,
+    econCharterKit,
+    retiredInstances,
+  ] = await Promise.all([
     installP,
     counterP,
+    previousInstanceP,
+    contractKitsP,
+    econCharterKitP,
+    provideRetiredInstances(retiredContractInstancesP, produceRetiredInstances),
   ]);
-  const terms = await harden({
-    binaryVoteCounterInstallation: counterInstall,
-  });
+
+  const label = 'econCommitteeCharter';
+  const previousCharterKit = { label, ...econCharterKit };
```

Comment:
This reads as if any label in `econCharterKit` overrides the one on line 335.

```suggestion
  const previousCharterKit = { ...econCharterKit, label };
```

Reply:
In some ways I prefer the other order. We need to add this because the label wasn't added initially, not because we want to change it. If this proposal is re-used, there's no reason to change an existing label.

But I'll change it.